apps/dashboard/src/hooks/use-feature-flag.tsx

```diff
@@ -1,11 +1,21 @@
+import { FeatureFlags, FeatureFlagsKeysEnum, prepareBooleanStringFeatureFlag } from '@novu/shared';
 import { useFlags } from 'launchdarkly-react-client-sdk';
-import { FeatureFlagsKeysEnum, prepareBooleanStringFeatureFlag } from '@novu/shared';
 import { LAUNCH_DARKLY_CLIENT_SIDE_ID } from '../config';
 
 function isLaunchDarklyEnabled() {
   return !!LAUNCH_DARKLY_CLIENT_SIDE_ID;
 }
 
+export const useFeatureFlagMap = (defaultValue = false): FeatureFlags => {
+  const flags = useFlags();
+
+  return Object.keys(flags).reduce((acc: FeatureFlags, flag: string) => {
+    acc[flag as keyof FeatureFlags] = flags[flag] ?? defaultValue;
+
+    return acc;
+  }, {} as FeatureFlags);
+};
```

Comment:
this hook seems redundant, like you iterate over original feature flags and duplicate its value to another map... if there is no value just check for the `undefined` in that case...

fyi iterations in hooks should be memoized with `useMemo` from React...